src/app/features/courses/components/courses-item/courses-item.component.ts

```diff
@@ -1,8 +1,21 @@
-import { Component } from '@angular/core';
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { CoursesItemModel } from '../../models/courses-item.model';
 
 @Component({
   selector: 'app-courses-item',
   templateUrl: './courses-item.component.html',
   styleUrls: ['./courses-item.component.scss']
 })
-export class CoursesItemComponent {}
+export class CoursesItemComponent {
+
+  @Input() course: CoursesItemModel;
+  @Output() deleteCourse: EventEmitter<CoursesItemModel>;
+
+  constructor() {
+    this.deleteCourse = new EventEmitter();
```

Comment:
minor: You can assign event emitter above, and than remove unused constructor 
`@Output() deleteCourse: EventEmitter<CoursesItemModel> = new EventEmitter();`